src/main/java/org/osc/manager/ism/api/IsmSecurityGroupInterfaceApi.java

```diff
@@ -40,167 +41,210 @@
-				Root<SecurityGroupInterfaceEntity> r = query.from(SecurityGroupInterfaceEntity.class);
-				query.select(r);
-				return IsmSecurityGroupInterfaceApi.this.em.createQuery(query).getResultList();
-			}
-		});
-	}
-
-	private SecurityGroupEntity getSecurityGroupbyId(String sgId) throws Exception {
-		IsmSecurityGroupApi ismSecurityGroupApi = IsmSecurityGroupApi.create(this.vs, this.txControl, this.em);
-		SecurityGroupEntity result = ismSecurityGroupApi.getSecurityGroup(sgId);
-		if (result == null) {
-			throw new Exception(String.format("Security Group with %s not found", sgId));
-		}
-		return result;
-	}
-
-	private Set<PolicyEntity> getPoliciesById(SecurityGroupInterfaceElement sgiElement) throws Exception {
-		Set<PolicyEntity> policies = new HashSet<>();
-		IsmPolicyApi ismPolicyApi = IsmPolicyApi.create(this.mc, this.txControl, this.em);
-		for (ManagerPolicyElement mgrPolicyElement : sgiElement.getManagerPolicyElements()) {
-			policies.add(ismPolicyApi.getPolicy(mgrPolicyElement.getId(), mgrPolicyElement.getDomainId()));
-		}
-		if (policies.isEmpty()) {
-			throw new Exception("No policies found");
-		}
-		return policies;
-	}
-
-	@Override
-	public void close() {
-		LOGGER.info("Closing connection to the database");
-		this.txControl.required(() -> {
-			this.em.close();
-			return null;
-		});
-	}
+    private static Logger LOGGER = Logger.getLogger(IsmSecurityGroupInterfaceApi.class);
+    private static String SGI_NOT_FOUND_MESSAGE = "A security group interface with id %s was not found.";
+    private TransactionControl txControl;
+    private EntityManager em;
+    private VirtualSystemElement vs;
+    private ApplianceManagerConnectorElement mc;
+    private IsmDeviceApi api;
+
+    public IsmSecurityGroupInterfaceApi(VirtualSystemElement vs, ApplianceManagerConnectorElement mc,
+            TransactionControl txControl, EntityManager em) {
+        // Set vs as class variable as needed
+        this.txControl = txControl;
+        this.em = em;
+        this.vs = vs;
+        this.mc = mc;
+    }
+
+    public static IsmSecurityGroupInterfaceApi create(VirtualSystemElement vs, ApplianceManagerConnectorElement mc,
+            TransactionControl txControl, EntityManager em) throws Exception {
+        return new IsmSecurityGroupInterfaceApi(vs, mc, txControl, em);
+    }
+
+    @Override
+    public String createSecurityGroupInterface(SecurityGroupInterfaceElement sgiElement) throws Exception {
+        return createSecurityGroupInterface(this.vs.getMgrId(), sgiElement);
+    }
+
+    public String createSecurityGroupInterface(String deviceId, SecurityGroupInterfaceElement sgiElement)
+            throws Exception {
+
+        this.api = new IsmDeviceApi(null, this.txControl, this.em);
+        DeviceEntity dev = (DeviceEntity) this.api.getDeviceById(deviceId);
+        if (dev == null) {
+            String msg = String.format("Cannot find the device id: %s\n", deviceId);
```

Comment:
nline in the end on purpose?

Reply:
yes and before starting new method newline is inserted. I am following as done in rest of the code base.